operator/v1/types_console.go

```diff
@@ -149,6 +149,9 @@ type DeveloperConsoleCatalogCustomization struct {
 	// +kubebuilder:validation:Optional
 	// +optional
 	Categories []DeveloperConsoleCatalogCategory `json:"categories,omitempty"`
+	// disabledTypes is a list of developer catalog IDs that are not shown to users.
+	// +optional
+	DisabledTypes []string `json:"disabledTypes,omitempty"`
```

Comment:
If I'm correct we agreed on @JoelSpeed idea to add an `enabledTypes` as well, or?

We can maybe also add a hint where to find the types = sub-catalog IDs. Wdyt?

```suggestion
	// enabledTypes is a list of developer catalog types (sub-catalogs IDs) that will be shown to users.
	// Sub-catalogs are added via console plugins, the available types (sub-catalog IDs) are available
	// in the console on the cluster configuration page, or when editing the YAML in the console.
	// +optional
	EnabledTypes []string `json:"enabledTypes,omitempty"`
	// disabledTypes is a list of developer catalog types (sub-catalogs IDs) that are not shown to users.
	// Sub-catalogs are added via console plugins, the available types (sub-catalog IDs) are available
	// in the console on the cluster configuration page, or when editing the YAML in the console.
	// +optional
	DisabledTypes []string `json:"disabledTypes,omitempty"`
```

Reply:
> obvious to an end user?

Hopefully, at least the term is used in our UI and in our frontend code. Fyi:

![image](https://user-images.githubusercontent.com/139310/188724918-16e83755-3dc8-4f78-9950-ed69007c6ad2.png)